src/shims/posix/thread.rs

```diff
@@ -14,7 +14,7 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         let this = self.eval_context_mut();
 
         this.tcx.sess.warn(
-            "thread support is experimental and incomplete: weak memory effects are not emulated.",
+            "thread support is experimental: weak memory effects are not fully compatible with the Rust atomics memory model.",
```

Comment:
I am honestly wondering if we should not just remove the warning. Yes, we don't emulate *all* weak memory effects, but we are doing a *lot* now and I think for the remaining issues it is enough to just document in the README that
- some real weak behaviors can never be observed in Miri (maybe with a list of the sitautions where this can happen)
- some weak behaviors have been forbidden by a change in C++20, but they can still happen in Miri since no known algorithm exists that is correct wrt C++20 and also supports sequentially consistent accesses

Reply:
Hm, thinking about it a bit more, we should probably keep the warning because of our non-preemptive scheduler. So maybe phrase this as:
> thread support is experimental: the scheduler is not preemptive, and can get stuck in spin loops.\n(see https://github.com/rust-lang/miri/issues/1388)

Reply:
Should I remove the warning now that the scheduler is preemptive?

Reply:
It is still not preemptive, I just made it properly round-robin (rather than giving threads with lower index priority).

I hope to make it preemptive in the (near) future, but I'll take care of the warning then.